### ISSUE TITLE
Port value for Zipkin Tracing should be not defaulted

### ISSUE THREAD

**Issue by arjav-desai** [oracle/helidon#3948]
ZipkinTracerBuilder created based on config, as specified in https://helidon.io/docs/latest/#/mp/tracing/02_zipkin, defaults port to 9411, if not specified.

As the config for host allows for FQDN, port should not be defaulted. I am running into issue pushing/sending Traces to OCI APM service since the collection/ingestion endpoint for the same does not require/specify port https://docs.oracle.com/en-us/iaas/application-performance-monitoring/doc/configure-open-source-tracing-systems.html#APMGN-GUID-B5EDE254-C854-436D-B844-B986A4E077AA (which will be case when user is using Zipkin as service, rather than hosting it).

**Comment by barchetta** [comment]
In theory this should work by specifying a port number in the config that matches the default port number for the protocol (80 for http or 443 for https). But apparently Arjav ran into a service that rejects the HTTP request in this case -- maybe because the service was not expecting a port number in the `Host:` header.

We do not want to change the default behavior because that would break compatibility. A couple options instead:
1.  Interpret `-1` for `port` to mean do not use a port number in the request URL.
2. Add a new property, like `noPort` to indicate no port number

The first option is appealing because it's providing a work-around for a bug in another product without inventing something new. And it is consistent with `URL.getPort()`

**Comment by tomas-langer** [comment]
Please specify port that is valid for your address - if you specify `https://zipkin.myserver.com`, the port should be `443`. Using default port for a protocol is not supported, as Zipkin does not use `80` or `443` by default (in docker/k8s installations) - so we decided to use the default port of zipkin installation (which is indeed `9411`). 

TL;DR:
The default is correct - Zipkin uses 9411 as the default port. If you need to use a different port, please specify it explicitly.

**Comment by tomas-langer** [comment]
Closing as this is the designed and intended behavior.

**Comment by barchetta** [comment]
I think there might still be in issue here. We should make sure specifying "-1" as a port number translates to "no port number" and we should make sure that specifying a collector URL with no port number preserves the URL with no port number. Specifically, this code looks suspicious: https://github.com/oracle/helidon/blob/799ebd22ceae7019b7a903b482c35635264615e3/tracing/tracing/src/main/java/io/helidon/tracing/TracerBuilder.java#L163

**Comment by barchetta** [comment]
I create #3966 for the issue described in the previous comment.
